content/techniques/queues.md

```diff
@@ -6,15 +6,445 @@ Queues are a powerful design pattern that help you deal with common application
+
+```typescript
+const job = await this.audioQueue.add(
+  'transcode',
+  {
+    foo: 'bar',
+  },
+  { delay: 3000 }, // 3 seconds delayed
+);
+```
+
+To add a job to the right end of the queue (process the job as **LIFO** (Last In First Out)), set the `lifo` property of the configuration object to `true`.
+
+```typescript
+const job = await this.audioQueue.add(
+  'transcode',
+  {
+    foo: 'bar',
+  },
+  { lifo: true },
+);
+```
+
+To prioritize a job, use the `priority` property.
+
+```typescript
+const job = await this.audioQueue.add(
+  'transcode',
+  {
+    foo: 'bar',
+  },
+  { priority: 2 },
+);
+```
+
+For a full list of options, check the API documentation [here](https://api.docs.bullmq.io/types/v4.JobsOptions.html) and [here](https://api.docs.bullmq.io/interfaces/v4.BaseJobOptions.html).
+
+#### Consumers
+
+A consumer is a **class** defining methods that either process jobs added into the queue, or listen for events on the queue, or both. Declare a consumer class using the `@Processor()` decorator as follows:
+
+```typescript
+import { Processor } from '@nestjs/bullmq';
+
+@Processor('audio')
+export class AudioConsumer {}
+```
+
+> info **Hint** Consumers must be registered as `providers` so the `@nestjs/bullmq` package can pick them up.
+
+Where the decorator's string argument (e.g., `'audio'`) is the name of the queue to be associated with the class methods.
+
+```typescript
+import { Processor, WorkerHost } from '@nestjs/bullmq';
+import { Job } from 'bullmq';
+
+@Processor('audio')
+export class AudioConsumer extends WorkerHost {
+  async process(job: Job<any, any, string>): Promise<any> {
+    let progress = 0;
+    for (i = 0; i < 100; i++) {
+      await doSomething(job.data);
+      progress += 1;
+      await job.progress(progress);
+    }
+    return {};
+  }
+}
+```
+
+The process method is called whenever the worker is idle and there are jobs to process in the queue. This handler method receives the `job` object as its only argument. The value returned by the handler method is stored in the job object and can be accessed later on, for example in a listener for the completed event.
+
+`Job` objects have multiple methods that allow you to interact with their state. For example, the above code uses the `progress()` method to update the job's progress. See [here](https://api.docs.bullmq.io/classes/v4.Job.html) for the complete `Job` object API reference.
+
+In the older version, Bull, you could designate that a job handler method will handle **only** jobs of a certain type (jobs with a specific `name`) by passing that `name` to the `@Process()` decorator as shown below.
+
+> warning **Warning** This doesn't work with BullMQ, keep reading.
+
+```typescript
+@Process('transcode')
+async transcode(job: Job<unknown>) { ... }
+```
+
+This behavior is not supported in BullMQ due to confusions it generated. Instead, you need switch cases to call different services or logic for each job name:
```

Comment:
I think its worth linking to the `bullmq` docs page where they say this 👍 : https://docs.bullmq.io/patterns/named-processor

I myself only ran into this when migrating my app from `@nestjs/bull` to `@nestjs/bullmq` and thought that it was a choice by the NestJS maintainers.

Reply:
It's linked on line 275, few lines below ;)